crates/c-api/src/wasi.rs

```diff
@@ -114,9 +109,11 @@ impl wasi_config_t {
                 builder.inherit_stdout();
             }
             WasiConfigWritePipe::File(file) => {
-                let file = cap_std::fs::File::from_std(file);
-                let file = wasi_common::sync::file::File::from_cap_std(file);
-                builder.stdout(Box::new(file));
+                let file = tokio::fs::File::from_std(file);
+                let stdout_stream = wasmtime_wasi::AsyncStdoutStream::new(
+                    wasmtime_wasi::pipe::AsyncWriteStream::new(1024 * 1024, file),
```

Comment:
I think the previous version didn't buffer things here, so could this keep the unbuffered versions?